yarn-project/p2p/src/services/peer-manager/peer_manager.ts

```diff
@@ -276,18 +322,36 @@ export class PeerManager {
    */
   private prioritizePeers(connections: Connection[]): Connection[] {
     if (connections.length > this.config.maxPeerCount) {
-      // Sort the peer scores from lowest to highest
-      const prioritizedConnections = connections.sort((connectionA, connectionB) => {
+      // First, separate trusted peers from regular peers
```

Comment:
in discover the remaining connections are calculated as:
    const healthyConnections = this.prioritizePeers(this.pruneUnhealthyPeers(this.pruneDuplicatePeers(connections)));
    
    
Instead of returning the trusted connections at the top of the list, we can simply it by not returning the trusted connections at all!

peersToConnect can be adjusted to:
```
const peersToConnect = this.config.mexPeerCount - healthyConnections.length - trustedPeers.length;
```